Pick OMI packages from omi-kits repo

vimish opened the pull request:
@Microsoft/ostc-devs 
@lagalbra

scx will no longer build omi packages. Instead it will
take the omi packages from the omi-kits repo where omi
team will check in latest certified omi packages for all
platforms.

jeffaco commented on build/Makefile.kits:
```diff
@@ -213,11 +229,17 @@ else # ifneq ($(COMBINED_PACKAGES),1)
+	  cd $(INTERMEDIATE_DIR)/098; mv omi-*.deb `ls omi-*.deb | sed "s/\.ssl_098\./\./g"`
+	  cd $(INTERMEDIATE_DIR)/098; mv omi-*.rpm `ls omi-*.rpm | sed "s/\.ssl_098\./\./g"`
+	  cd $(INTERMEDIATE_DIR)/100; mv omi-*.deb `ls omi-*.deb | sed "s/\.ssl_100\./\./g"`
+	  cd $(INTERMEDIATE_DIR)/100; mv omi-*.rpm `ls omi-*.rpm | sed "s/\.ssl_100\./\./g"`
```
Blech. I get what you're doing. I debated other options for naming, but we ended up zeroing in on what we did. You could have probably left that in the name if the shell bundle dealt with it properly too, although it doesn't matter much.

jeffaco commented on build/Makefile.kits:
```diff
@@ -213,11 +229,17 @@ else # ifneq ($(COMBINED_PACKAGES),1)
 	# (Copying for combined packages)
         ifeq ($(DISABLE_LISTENER),0)
```
Shouldn't the `DISABLE_LISTENER` stuff come out too?

I think that it shouldn't be based on installbuilder flags (like it is now), nor on the existence of a marker file to NOT enable the listener. Instead, we should never enable the listener unless a marker file is created to do so. This can be controlled by the customer (create the marker file before installing the package), or by a new qualifier to the shell-bundle (`--enable-om-port` or something).

So I'd recommend eliminating line 230, fixing indentation under it, removing definition of DISABLE_LISTENER when invoking installbuilder, then make the change to the shell bundle (and post installation scripts) appropriately.

I'll mark this waiting so I see those changes go in. Obviously, you'll need to touch the MPs on Windows to do whatever it needs (run with --enable-om-port, or create the marker file before installing the package, or whatever. That will be separate to this since that stuff still lives in TFS anyway.

vimish replied:
It seems DISABLE_LISTENER is passed to create_bundle as value for provider only build. We are planning to remove provider only option. I will remove all references to DISABLE_LISTENER in that commit.

jeffaco commented on installer/datafiles/Base_SCXCore.data:
```diff
@@ -207,15 +207,13 @@ ConfigureRunAs() {
+    if [ -f /etc/scxagent-enable-port ]; then
         # Add port 1270 to the list of ports that OMI will listen on
         /opt/omi/bin/omiconfigeditor httpsport -a 1270 < /etc/opt/omi/conf/omiserver.conf > /etc/opt/omi/conf/omiserver.conf_temp
         mv /etc/opt/omi/conf/omiserver.conf_temp /etc/opt/omi/conf/omiserver.conf
```
These lines should be indented back to line up with the others (208, etc).

vimish commented:
@jeffaco @lagalbra 
Added check to purge scx directories only when scx package is removed.

Updated .travis.yml to build from my branch. Will remove this before merging.

vimish commented on build/Makefile:
```diff
@@ -15,6 +15,7 @@ PF_POSIX=true
 # Get the root of the SCXPAL directory (and other easy locations)
 SCXPAL_DIR := $(shell cd ../../pal; pwd)
 SCXOMI_DIR := $(shell cd ../../omi/Unix; pwd -P)
+OMIKITS_DIR := $(shell cd ../../omi-kits/release; pwd)
```
Build-SCXcore (https://github.com/Microsoft/Build-SCXcore)

jeffaco commented on installer/bundle/bundle_skel_Linux.sh:
```diff
@@ -332,11 +333,11 @@ remove_and_install()
     check_if_pkg_is_installed apache-cimprov
     if [ $? -eq 0 ]; then
-        pkg_rm apache-cimprov force
+        pkg_rm apache-cimprov
```
Currently the code is written to be "generic" (i.e. "docker-cimprov, apache-cimprov, mysql-cimprov").

This list doesn't change much; you could hard-code it, but you could also do something more generic based on what is installed (i.e. *-cimprov), processing the list. I'll leave that up to you.

Also be aware the SCX installs an uninstall script that can be run by the shell bundle to remove software. The intent was that the uninstall script (`/opt/microsoft/scx/bin/uninstall`) was the "latest/greatest" way to install software, and doing removal in the shell bundle was mostly for legacy versions if that file didn't exist.

So if you change the way to remove software, you should modify the uninstall script as well.

jeffaco commented on installer/bundle/bundle_skel_Linux.sh:
```diff
@@ -513,10 +521,13 @@ then
+        check_if_pkg_is_installed scx
+        if [ $? -ne 0 ]; then
+            echo "Purging all files in cross-platform agent ..."
+            rm -rf /etc/opt/microsoft/*-cimprov /etc/opt/microsoft/scx /opt/microsoft/*-cimprov /opt/microsoft/scx /var/opt/microsoft/*-cimprov /var/opt/microsoft/scx
```
You shouldn't be nuking the *-cimprov directories unless underlying packages are removed.

So don't nuke .../apache-cimprov unless the apache package is removed. Same for MySQL.

vimish commented:
@jeffaco 
Added check to purge directories based on the provider it belongs to.

I will update the uninstall script to remove the packages in another commit.

lagalbra commented on installer/bundle/bundle_skel_Linux.sh:
```diff
@@ -513,10 +521,23 @@ then
-        rm -rf /etc/opt/microsoft/*-cimprov /etc/opt/microsoft/scx /opt/microsoft/*-cimprov /opt/microsoft/scx /var/opt/microsoft/*-cimprov /var/opt/microsoft/scx
-        rmdir /etc/opt/microsoft /opt/microsoft /var/opt/microsoft 1>/dev/null 2>/dev/null
-
+        check_if_pkg_is_installed apache-cimprov
```
Adding onto Jeff's comment about keeping these provider references generic, perhaps these two -cimprov checks can be eliminated and combined into the scx check like the OMS code does [here](https://github.com/Microsoft/OMS-Agent-for-Linux/blob/master/installer/scripts/uninstall.sh#L109)

lagalbra commented on installer/bundle/bundle_skel_Linux.sh:
```diff
@@ -513,10 +521,23 @@ then
+            echo "Purging all files in mysql provider ..."
+            rm -rf /etc/opt/microsoft/mysql-cimprov /opt/microsoft/mysql-cimprov /var/opt/microsoft/mysql-cimprov
+        fi
+        # Remove directoried only if scx got removed successfully (Other products might be dependent on scx)
```
nit: spelling: directories
